tools/layer-zero-example/contracts/ExampleOFTHTS.sol

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./hts/HederaTokenService.sol";
+import "./hts/IHederaTokenService.sol";
+import "./hts/KeyHelper.sol";
+import "./hts/ExpiryHelper.sol";
+import "./BaseHTSOFT.sol";
+
+contract ExampleOFTHTS is Ownable, KeyHelper, ExpiryHelper, HederaTokenService, BaseHTSOFT {
+    address public htsTokenAddress;
+
+    event CreatedToken(address tokenAddress);
+
+    constructor(
+        string memory _name,
+        string memory _symbol,
+        address _lzEndpoint,
+        address _delegate
+    ) payable BaseHTSOFT(_name, _symbol, _lzEndpoint, _delegate) Ownable(_delegate) {
+        IHederaTokenService.TokenKey[] memory keys = new IHederaTokenService.TokenKey[](2);
+        keys[0] = getSingleKey(
+            KeyType.ADMIN,
+            KeyType.PAUSE,
+            KeyValueType.INHERIT_ACCOUNT_KEY,
+            bytes("")
+        );
+        keys[1] = getSingleKey(
+            KeyType.SUPPLY,
+            KeyValueType.INHERIT_ACCOUNT_KEY,
+            bytes("")
+        );
+
+        IHederaTokenService.Expiry memory expiry = IHederaTokenService.Expiry(0, address(this), 8000000);
+        IHederaTokenService.HederaToken memory token = IHederaTokenService.HederaToken(
+            _name, _symbol, address(this), "memo", true, 1000, false, keys, expiry
+        );
+
+        (int responseCode, address tokenAddress) = HederaTokenService.createFungibleToken(
+            token, 1000, int32(int256(uint256(8)))
+        );
+        require(responseCode == HederaResponseCodes.SUCCESS, "Failed to create HTS token");
+
+        htsTokenAddress = tokenAddress;
+
+        emit CreatedToken(tokenAddress);
+    }
+
+    /**
+     * @dev Burns tokens from the sender's specified balance.
+     * @param _from The address to debit the tokens from.
+     * @param _amountLD The amount of tokens to send in local decimals.
+     * @param _minAmountLD The minimum amount to send in local decimals.
+     * @param _dstEid The destination chain ID.
+     * @return amountSentLD The amount sent in local decimals.
+     * @return amountReceivedLD The amount received in local decimals on the remote.
+     */
+    function _debit(
+        address _from,
+        uint256 _amountLD,
+        uint256 _minAmountLD,
+        uint32 _dstEid
+    ) internal virtual override returns (uint256 amountSentLD, uint256 amountReceivedLD) {
+        (amountSentLD, amountReceivedLD) = _debitView(_amountLD, _minAmountLD, _dstEid);
+
+        address spender = _msgSender();
+        if (_from != spender) {
+            int256 response = HederaTokenService.approve(htsTokenAddress, spender, amountSentLD);
+            require(response == HederaResponseCodes.SUCCESS, "HTS: Approve failed");
+        }
+
```

Comment:
Isnt the transferFrom missing to be able to make a burn?

       int256 transferResponse = HederaTokenService.transferToken(htsTokenAddress,from,address(this) int64(uint64(_amountLD)));
        require(transferResponse == HederaResponseCodes.SUCCESS, "HTS: Transfer failed");